lib/rules/no-restricted-exports.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * @fileoverview Restrict usage of specified export names.
+ * @author Bradley Farias
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const astUtils = require("../ast-utils");
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        docs: {
+            description: "disallow usage of specified names when using `export`.",
+            category: "ECMAScript 6",
+            recommended: true
+        },
+    },
+
+    create(context) {
+        const sourceCode = context.getSourceCode();
+        const names = new Set([
```

Comment:
How about adding the full set of Object.prototype members?  I doubt that any good can come from silently allowing them.

"__proto__",
"__defineGetter__",
"__defineSetter__",
"__lookupGetter__",
"__lookupSetter__",
"hasOwnProperty",
"isPrototypeOf",
"propertyIsEnumerable",
"toLocaleString"